### PR TITLE
Implement adding/removing labels to/from repositories and images API

### PR DESCRIPTION
**Pull request by ywk253100**

### CODE REVIEW

File: docs/swagger.yaml [hunk 1]
```diff
@@ -995,6 +995,82 @@ paths:
...
+          required: true
+          description: The name of repository.
+      tags:
+        - Products
+      responses:
+        '200':
+          description: Successfully.
+        '401':
+          description: Unauthorized.
+        '403':
```
Comment (reasonerjt) [inline docs/swagger.yaml]: It would be helpful to specify what permission is needed under `403`

File: make/photon/db/registry.sql [hunk 2]
```diff
@@ -272,6 +272,20 @@ create table harbor_label (
  CONSTRAINT unique_name_and_scope UNIQUE (name,scope)
  );
 
+create table harbor_resource_label (
+ id int NOT NULL AUTO_INCREMENT,
+ label_id int NOT NULL,
+ resource_id varchar(256) NOT NULL,
```
Comment (reasonerjt) [inline make/photon/db/registry.sql]: Could you specify what the `resource_id` would look like for different `resource_type`?

File: docs/swagger.yaml [hunk 3]
```diff
@@ -1075,6 +1151,97 @@ paths:
               $ref: '#/definitions/DetailedTag'
         '500':
           description: Unexpected internal errors.
+  '/repositories/{repo_name}/tags/{tag}/labels':
+    get:
```
Comment (reasonerjt) [inline docs/swagger.yaml]: schema of the response?

File: docs/swagger.yaml [hunk 4]
```diff
@@ -1075,6 +1151,97 @@ paths:
...
+          in: path
+          type: string
+          required: true
+          description: The name of repository.
+        - name: tag
+          in: path
+          type: string
+          required: true
+          description: The tag of the image.
+        - name: label_id
```
Comment (reasonerjt) [inline docs/swagger.yaml]: This is the `resource_label_id` right? not the `label_id`.

File: src/ui/router.go [hunk 5]
```diff
@@ -70,7 +70,11 @@ func initRouters() {
 	beego.Router("/api/repositories", &api.RepositoryAPI{}, "get:Get")
 	beego.Router("/api/repositories/scanAll", &api.RepositoryAPI{}, "post:ScanAll")
 	beego.Router("/api/repositories/*", &api.RepositoryAPI{}, "delete:Delete;put:Put")
+	beego.Router("/api/repositories/*/labels", &api.RepositoryLabelAPI{}, "get:GetOfRepository;post:AddToRepository")
+	beego.Router("/api/repositories/*/labels/:id([0-9]+", &api.RepositoryLabelAPI{}, "delete:RemoveFromRepository")
```
Comment (reasonerjt) [inline src/ui/router.go]: there seems an extra `(`?
same with line 77

File: docs/swagger.yaml (hunk 4 above)
Comment (ywk253100) [reply docs/swagger.yaml]: It's `label_id`.

File: src/ui/api/repository_label.go [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+		r.HandleInternalServerError(fmt.Sprintf("failed to add label %d to resource %s %s: %v",
+			rl.LabelID, rl.ResourceType, rl.ResourceID, err))
+		return
+	}
+
+	// return the ID of label and return status code 200 rather than 201 as the label is not created
+	r.Redirect(http.StatusOK, strconv.FormatInt(rl.LabelID, 10))
+}
+
+func (r *RepositoryLabelAPI) removeLabel(rl *models.ResourceLabel) {
```
Comment (reasonerjt) [inline src/ui/api/repository_label.go]: The resource_label should be removed if the label is deleted.

File: docs/swagger.yaml (hunk 4 above)
Comment (reasonerjt) [reply docs/swagger.yaml]: I don't fully agree but given this is how github works we can leave this implement as-is.

File: src/ui/api/repository_label.go (hunk 6 above)
Comment (ywk253100) [reply src/ui/api/repository_label.go]: Open #4410 to track this, will submit another PR to fix it.

**Pull request merged**